vscode/e2e/utils/vscody/fixture.ts

```diff
@@ -310,7 +453,62 @@ const implFixture = _test.extend<TestContext, WorkerContext>({
                 adapters: ['node-http'],
                 recordFailedRequests: true,
                 logLevel: 'SILENT',
-                matchRequestsBy: defaultMatchRequestsBy,
+                timing: Timing.relative(1.0), //TODO: Configuration for fuzzy/flake testing
+                matchRequestsBy: {
+                    //TODO: Think of a clever way that we can require order on some types of requests
+                    order: false,
+                    method: true,
+                    url(url, req) {
+                        const parsed = new URL(url)
+                        parsed.searchParams.delete('client-version')
+                        const mitmProxy = getFirstOrValue(req.headers[MITM_PROXY_SERVICE_NAME_HEADER])
+                        if (mitmProxy) {
+                            parsed.hostname = `${mitmProxy}.proxy`
+                            parsed.port = ''
+                            parsed.protocol = 'http:'
+                        }
+                        //todo: replace host with semantic name if available
+                        return parsed.toString()
+                    },
+                    // Canonicalize JSON bodies so that we can replay the recording even if the JSON strings
+                    // differ by semantically meaningless things like object key enumeration order.
+                    body(body, req) {
+                        const contentType = getFirstOrValue(req.getHeader('content-type'))?.toLowerCase()
+                        if (contentType === 'application/json') {
+                            return jsonStableStringify(JSON.parse(body))
+                        }
+                        //TODO: Remove client version variables
+                        if (!contentType && typeof body === 'string') {
+                            const trimmedBody = body.trim()
+                            if (
+                                (trimmedBody.startsWith('{') && trimmedBody.endsWith('}')) ||
+                                (trimmedBody.startsWith('[') && trimmedBody.endsWith(']'))
+                            ) {
+                                let json: any = null
+                                try {
+                                    //only allow this to fail silently
+                                    json = JSON.parse(trimmedBody)
+                                } catch {}
```

Comment:
shouldn't we be returning the original body in this case?
```suggestion
                                } catch { return body }
```